compiler/rustc_lint/src/builtin.rs

```diff
@@ -2595,7 +2596,7 @@ declare_lint! {
 }
 
 pub struct ClashingExternDeclarations {
-    seen_decls: FxHashMap<Symbol, HirId>,
+    seen_decls: FxHashMap<String, HirId>,
```

Comment:
In theory this could just store `&'tcx str` without an issue, but `impl_lint_pass` does not support lints parametric over a lifetime T_T.

Reply:
> could just store &'tcx str without an issue

Yeah, you're right -- that would let me avoid the allocation completely.

> the impl_lint_pass! macro does not support lints parametric over a lifetime T_T

I don't think I'll address this in this PR --  I've just added a FIXME note for now,  until `impl_lint_pass!` can take lifetimes.